File: 2-regresion-lineal-simple/Regresion_lineal_simple.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv("Salary_Data.csv")
x = dataset.iloc[:,:-1].values
y = dataset.iloc[:,1].values

from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size = 1/3,random_state = 0)

#Crear modelo de Regresión lineal Simple con el conjunto de entrenamiento
from sklearn.linear_model import LinearRegression
regression = LinearRegression()
regression.fit(x_train,y_train)

#predecir el conjunto de test
y_pred = regression.predict(x_test)

#visualizar los resultados de entrenamiento
plt.scatter(x_train,y_train,color = "red")
plt.plot(x_train,regression.predict(x_train),color = "blue")
plt.title("Sueldo vs años de experiencia")
plt.xlabel("años de trabajo")
plt.ylabel("sueldo")
plt.show()

#visualizar los resultados de test
plt.scatter(x_test,y_test,color = "red")
plt.plot(x_train,regression.predict(x_train),color = "blue")
plt.title("Sueldo vs años de experiencia")
plt.xlabel("años de trabajo")
plt.ylabel("sueldo")
plt.show()

File: 2-regresion-lineal-simple/Regresion_lineal_simple.R
#Regresion Lineal simple

dataset = read.csv("Salary_Data.csv")

install.packages("caTools")
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary,SplitRatio = 2/3)
training_set = subset(dataset,split == TRUE)
testing_set = subset(dataset,split == FALSE)

# Ajustar el modelo de regresion lineal simple
regressor = lm(formula = Salary ~ YearsExperience,data = training_set)

#Predecir resultados con el conjunto de test
y_pred = predict(regressor,newdata = testing_set)

#visualizacion de los resultados 
install.packages("ggplot2")
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience,y = training_set$Salary),colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), colour = "blue") +
  ggtitle("Sueldo vs años de experiencia") +
  xlab("Años de experiencia") +
  ylab("Sueldo")

ggplot() +
  geom_point(aes(x = testing_set$YearsExperience,y = testing_set$Salary),colour = "red") +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), colour = "blue") +
  ggtitle("Sueldo vs años de experiencia") +
  xlab("Años de experiencia") +
  ylab("Sueldo")